recipes/global_minimum.py

```diff
@@ -0,0 +1,167 @@
+    """
+    Find the global minimum of the ligand (RDKit UFF or user-defined PLAMS |Job|) by systematically varying dihedral angles within the molecule.
+
+    ''mol'' |Molecule|: The input molecule
+    ''n_scans'' int: The number of times the global minimum search should be repeated
+    ''no_h'' bool: If hydrogen-containing bonds should ignored
+    ''no_ring'' bool: If bonds in ring systems should ignored
+    ''bond_orders'' list: A list of accepted bond orders (floats); a bond will be ignored if its bond order is not in *bond_orders*
+    ''job_type'' ''False'' or type: A type object of a class derived from |Job|. Set to ''False'' to use RDKit UFF
+    ''path'' str: The path where the PLAMS working directory will be stored
+    ''kwarg'' dict: Keyword arguments for *job_type*
+    ''return'' |Molecule|: A copy of *mol* with a newly optimized geometry
+    """
+    # Creates guess bonds if no bonds are present
+    if len(mol.bonds) == 0:
+        mol.guess_bonds()
+
+    # Create a list of 2-tuples (i.e. atomic indices) representing (valid) bonds within the molecule
+    bond_list = find_bond(mol, no_h=no_h, no_ring=no_ring, bond_orders=bond_orders)
+
+    # Search for the global minimum with RDKit UFF or with PLAMS at an user-defined level of theory
+    if not job_type:
+        if 'rdkit.Chem.AllChem' not in sys.modules or 'rdkit.Chem.rdForceFieldHelpers' not in sys.modules:
+            raise ModuleNotFoundError('rdkit.chem module not found, aborting RDKit UFF optimization')
+        if not no_ring:
+            raise TypeError('no_ring=False is not supported in combination with RDKit UFF')
+        if not rdForceFieldHelpers.UFFHasAllMoleculeParams(to_rdmol(mol)):
+            raise ValueError('No UFF parameters found for one or more atoms')
+
+        for i in range(n_scans):
+            for bond in bond_list:
+                mol = global_minimum_scan_rdkit(mol, bond)
+
+        # Optimize the molecule even if no valid bonds are found
+        if not bond_list:
+            rdmol = to_rdmol(mol)
+            AllChem.UFFGetMoleculeForceField(rdmol).Minimize()
+            mol = from_rdmol(rdmol)
+
+    else:
+        init(path=path)
+        for i in range(n_scans):
+            for bond in bond_list:
+                mol = global_minimum_scan_plams(mol, bond, job_type, **kwarg)
+
+        # Optimize the molecule even if no valid bonds are found
+        if not bond_list:
+            job = job_type(molecule=mol, **kwarg)
+            results = job.run()
+            mol = results.get_main_molecule()
+        finish()
+
+    return mol
+
+
+def find_bond(mol, no_h=True, no_ring=True, bond_orders=[1.0]):
+    """
+    Create a list of bonds. Each entry is a tuple with indices of atoms forming a dihedral.
+    Consider only diherdals with axis being a single bond, so that rotation is possible.
+
+    ''mol'' |Molecule|: The input molecule
```

Comment:
Could you please use the ``` :param bool removeHs: ``` syntax everywhere in this file? Just to stay consistent with the rest?